app/src/test/java/com/example/concordia_campus_guide/ModelTests/BuildingTest.java

```diff
@@ -0,0 +1,265 @@
+    }
+
+    @Test
+    public void setAddress() {
+        assertEquals("1455 De Maisonneuve West", buildingWithArgs.getAddress());
+        buildingWithArgs.setAddress("1190 Guy Street");
+        assertEquals("1190 Guy Street", buildingWithArgs.getAddress());
+    }
+
+    @Test
+    public void getDepartments() {
+        assertEquals(departments, buildingWithArgs.getDepartments());
+    }
+
+    @Test
+    public void setDepartments() {
+        ArrayList<String> updateDepartments =  new ArrayList<String>(Arrays.asList("dep1", "dep2", "dep3", "dep4"));
+        assertEquals(departments, buildingWithArgs.getDepartments());
+        buildingWithArgs.setDepartments(updateDepartments);
+        assertEquals(updateDepartments, buildingWithArgs.getDepartments());
+    }
+
+    @Test
+    public void getServices() {
+        assertEquals(services, buildingWithArgs.getServices());
+    }
+
+    @Test
+    public void setServices() {
+        ArrayList<String> updateServices =  new ArrayList<String>(Arrays.asList("serv1", "serv2", "serv3"));
+        assertEquals(services, buildingWithArgs.getServices());
+        buildingWithArgs.setServices(updateServices);
+        assertEquals(updateServices, buildingWithArgs.getServices());
+    }
+
+    @Test
+    public void getCornerCoordinates() {
+        assertEquals(cornerCoordinates, buildingWithArgs.getCornerCoordinates());
+    }
+
+    @Test
+    public void setCornerCoordinates() {
+        List<Coordinates> updatedListOfCornerCoordinates = new ArrayList<Coordinates>() {{
+            add(cornerCoordinate1);
+            add(cornerCoordinate2);
+        }};
+        ListOfCoordinates updatedCornerCoordinates =  new ListOfCoordinates(updatedListOfCornerCoordinates);
+        assertEquals(cornerCoordinates, buildingWithArgs.getCornerCoordinates());
+        buildingWithArgs.setCornerCoordinates(updatedCornerCoordinates);
+        assertEquals(updatedCornerCoordinates, buildingWithArgs.getCornerCoordinates());
+    }
+
+    @Test
+    public void getServicesStringEmpty(){
+        buildingWithArgs.setServices(new ArrayList<>());
+        assertTrue(buildingWithArgs.getServices().isEmpty());
+        assertEquals("", buildingWithArgs.getServicesString());
+    }
+
+    @Test
+    public void getServicesString() {
+        assertFalse(buildingWithArgs.getServices().isEmpty());
+    }
+
+    @Test
+    public void getDepartmentsStringEmpty() {
+        buildingWithArgs.setDepartments(new ArrayList<>());
+        assertTrue(buildingWithArgs.getDepartments().isEmpty());
+        assertEquals("", buildingWithArgs.getDepartmentsString());
+    }
+
+    @Test
+    public void getDepartmentsString() {
+        assertFalse(buildingWithArgs.getDepartments().isEmpty());
+    }
+
+    @Test
+    public void getGeoJson() throws JSONException {
+        assertNotNull(buildingWithArgs.getGeoJson());
+//        JSONObject toReturn = buildingWithArgs.getGeoJson();
```

Comment:
why is this commented out? If it's not needed, please remove it eh

Reply:
@narrapangan